src/convert.cpp

```diff
@@ -161,22 +177,43 @@ ddwaf_object* to_ddwaf_object(
     bool boolValue = val.ToBoolean().Value();
     return ddwaf_object_bool(object, boolValue);
   }
-  if (val.IsArray()) {
```

Comment:
Not deleted, just moved to line 190, to have it in good order: checks -> Array -> Object and be able to check and add object to jsSet in a right place.